js/sparkswap.js

```diff
@@ -173,7 +173,50 @@ module.exports = class sparkswap extends Exchange {
     }
 
     async fetchMarkets () {
-        throw new ExchangeError ('Not Implemented');
+        let response = await this.privateGetV1Markets ();
+        let markets = response['supported_markets'];
+        let result = [];
+        for (let i = 0; i < markets.length; i++) {
+            const market = markets[i];
+            const id = market['id'];
+            const symbolParts = id.split ('/');
+            const baseId = symbolParts[0];
+            const quoteId = symbolParts[1];
+            const base = this.commonCurrencyCode (baseId.toUpperCase ());
+            const quote = this.commonCurrencyCode (quoteId.toUpperCase ());
+            const active = true;
+            const precision = {
+                'amount': market['amountPrecision'],
```

Comment:
Looks like these need to be integers instead of strings (which is what im assuming is the return value). We can use `this.safeInteger(market, 'amountPrecision')` here

Reply:
More info: https://github.com/ccxt/ccxt/wiki/Manual#market-structure